xgi/classes/hypergraph.py

```diff
@@ -668,6 +685,24 @@ def add_edges_from(self, ebunch_to_add, **attr):
             try:
                 e = next(new_edges)
             except StopIteration:
+
+                if format2 or format4:
+                    # If we don't set the start of self._edge_uid correctly, it will start at 0,
+                    # which will overwrite any existing edges when calling add_edge().  First, we
+                    # use the somewhat convoluted float(e).is_integer() instead of using
+                    # isinstance(e, int) because there exist integer-like numeric types (such as
+                    # np.int32) which fail the isinstance() check.
+                    edges_with_int_id = [
```

Comment:
My initial thought on this is that `add_edges()` is not covered - is there a reason for this? Also, perhaps we can add an `update_uid` function or something that evaluates if the ID to be added (in either `add_edges_from` or `add_edge`) is an int and larger than the current uid and if so, updates the edge uid.

Reply:
You mean `add_edge()` is not covered? I think it was already dealing correctly with new ids. 
The problem that I was fixing was that adding edges (without specifying their ids) was not updating the internal edge_uid. Because of this, any edge added after that was given uid 0 by default, replacing the existing edge with that uid. So the problem was not about the new uid, but about the current uid that was not correct. We could replace the current fix by a function sure.

Reply:
I think that the same is true for `add_edge()`. If you run
```
H = xgi.Hypergraph()
H.add_edge([1, 2], id=0)
H.add_edge([3, 4])
H.edges.members(dtype=dict)
{0: {3, 4}}
```
I don't think it increments the counter, right?

Reply:
It does increment it, the output is `{0: {1, 2}, 1: {3, 4}}` and the _edge_ui is `count(2)` :)

Reply:
Mmmmm. Fair. What about this one, though?
```
import xgi
H = xgi.Hypergraph()
H.add_edge([1, 2], id=1)
H.add_edge([3, 4])
H.add_edge([5, 6])
H.edges.members(dtype=dict)
```

Reply:
Ooh ok you got me, good one: output `{1: {5, 6}, 0: {3, 4}}` is wrong.